go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -81,69 +81,64 @@ func newEgressFirewallRule(rawEgressFirewallRule egressfirewallapi.EgressFirewal
 	return efr, nil
 }
 
-// This function is used to sync egress firewall setup. It does three "cleanups"
+// This function is used to sync egress firewall setup. It does two cleanups
 
-// - 	Cleanup the old implementation (using LRP) in local GW mode -> new implementation (using ACLs) local GW mode
+// - 	Cleanup the old local GW mode implementation (using LRP)
 //  	For this it just deletes all LRP setup done for egress firewall
 
-// -	Cleanup the new local GW mode implementation (using ACLs on the node switch) -> shared GW mode implementation (using ACLs on the join switch)
-//  	For this it just deletes all ACL setup done for egress firewall on the node switches
-
-// -	Cleanup the old implementation (using LRP) in local GW mode -> shared GW mode implementation (using ACLs on the join switch)
-//  	For this it just deletes all LRP setup done for egress firewall
+// -	Cleanup the old shared GW mode implementation (using ACLs on the all switches with direction=from-lport)
+//  	For this it just deletes all the ACL setup done for egress firewall on all switches with direction=from-lport
 
 // NOTE: Utilize the fact that we know that all egress firewall related setup must have a priority: types.MinimumReservedEgressFirewallPriority <= priority <= types.EgressFirewallStartPriority
 func (oc *Controller) syncEgressFirewall(egressFirwalls []interface{}) {
-	if config.Gateway.Mode == config.GatewayModeShared {
-		// Mode is shared gateway mode, make sure to delete all ACLs on the node switches
-		egressFirewallACLIDs, stderr, err := util.RunOVNNbctl(
-			"--data=bare",
-			"--no-heading",
-			"--columns=_uuid",
-			"--format=table",
-			"find",
-			"acl",
-			fmt.Sprintf("priority<=%s", types.EgressFirewallStartPriority),
-			fmt.Sprintf("priority>=%s", types.MinimumReservedEgressFirewallPriority),
-		)
+	egressFirewallACLIDs, stderr, err := util.RunOVNNbctl(
+		"--data=bare",
+		"--no-heading",
+		"--columns=_uuid",
+		"--format=table",
+		"find",
+		"acl",
+		fmt.Sprintf("priority<=%s", types.EgressFirewallStartPriority),
+		fmt.Sprintf("priority>=%s", types.MinimumReservedEgressFirewallPriority),
+		fmt.Sprintf("direction=%s", types.DirectionFromLPort),
```

Comment:
this works for removing all legacy "from-lport" rules on an upgrade to this new version, but what about future upgrades where you need to sync/cleanup old "to-port" rules?

Reply:
The current sync only syncs "legacy implementation state" to "new implementation state". Your concern falls under the category "egress firewall should sync NB-db and k8s state when ovn-kubernetes starts", this has not been implemented for egress firewall, any reasons as to why @JacobTanenbaum ?

Reply:
No just has not made it to the top of my list yet